### DISCUSSION TITLE
Developing a Plugin to Extend the Form Builder

### DISCUSSION THREAD

**Discussion by catmasik** [elementor/elementor#17945] [Developers Community]
Good day!

We are trying to develop an extension for Elementor that could add 
extended widgets to the Elementor Form Builder.
To develop form widgets, we inherited the parent class "ElementorPro\Modules\Forms\Fields\Field_Base".
But during development, we encountered some problems. When inserting our widget into the form, the 
widget is not immediately drawn in the form, although everything is fine when the page is reloaded.
We also tried to add controls to our widget that would allow us to change the field settings inside the form. 
To do this, we override the "add_control_section_to_form" method by adding our own controls. 
However, for some reason, these controls were added to all form fields, although we expected them to apply 
only to our field widget.
We kindly ask you to either give us a link to the documentation where we could study the question 
of how to write extensions for the Elementor Form Builder, or give us an example of how we can
 perform a similar task.

Thank you!

**Comment by aaliyahfraser** [comment]
I have also posted a blog on this topic. This is because Elementor has so many free amazing add-ons. Easy-to-use features made creating websites a breeze. Here we talk about “Best Elementor Addons and Plugins for WordPress 2023 – Compared” You can check out it at – https://quomodosoft.com/best-elementor-addons-and-plugins/
